worker/src/lib.rs

```diff
@@ -0,0 +1,592 @@
+        P::WorkData: 'static + Send,
+    {
+        unsafe {
+            let worker_data = mem::ManuallyDrop::new(worker_data);
+            let size = mem::size_of_val(&worker_data) as u32;
+            let ptr = &worker_data as *const _ as *const c_void;
+            let schedule_work = if let Some(schedule_work) = self.internal.schedule_work {
+                schedule_work
+            } else {
+                return Err(WorkerError::Unknown);
+            };
+            match (schedule_work)(self.internal.handle, size, ptr) {
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_SUCCESS => Ok(()),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_UNKNOWN => Err(WorkerError::Unknown),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_NO_SPACE => Err(WorkerError::NoSpace),
+                _ => Err(WorkerError::Unknown),
+            }
+        }
+    }
+}
+
+/// Handler available inside worker function to send response to `run()` context.
+///
+/// ResponseHandler need the current Worker trait implementor as generic parameter, because it use
+/// the `ResponseData` associated type to know the datatype to send to the `worker_response`
+/// method.
+pub struct ResponseHandler<P: Worker> {
+    /// function provided by the host to send response to `run()`
+    response_function: lv2_sys::LV2_Worker_Respond_Function,
+    /// Response handler provided by the host, must be passed to the host provided
+    /// response_function.
+    respond_handle: lv2_sys::LV2_Worker_Respond_Handle,
+    phantom: PhantomData<P>,
+}
+
+impl<P: Worker> ResponseHandler<P> {
+    /// Allow to give response to the `run` context.
+    ///
+    /// This method allow to give a response to the `run` context. After calling this method, the host will call
+    /// `worker_response` and give the passed data.
+    ///
+    /// **Notes about the passed data:** The buffer used to pass data is managed by the host. That
+    /// mean the size is unknown and may be limited. So if you need to pass huge amount of data,
+    /// it's preferable to use another way, for example a sync::mpsc channel.
+    pub fn respond(&self, response_data: P::ResponseData) -> Result<(), WorkerError>
+    where
+        P::WorkData: 'static + Send,
+    {
+        unsafe {
+            let response_data = mem::ManuallyDrop::new(response_data);
+            let size = mem::size_of_val(&response_data) as u32;
+            let ptr = &response_data as *const _ as *const c_void;
+            let response_function = self.response_function.ok_or(WorkerError::Unknown)?;
+            match (response_function)(self.respond_handle, size, ptr) {
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_SUCCESS => Ok(()),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_UNKNOWN => Err(WorkerError::Unknown),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_NO_SPACE => Err(WorkerError::NoSpace),
+                _ => Err(WorkerError::Unknown),
+            }
+        }
```

Comment:
Same here...

```suggestion
        let response_data = mem::ManuallyDrop::new(response_data);
        let size = mem::size_of_val(&response_data) as u32;
        let ptr = &response_data as *const ManuallyDrop<P::ResponseData> as *const c_void;
        let response_function = self.response_function.ok_or(WorkerError::Unknown)?;
        match unsafe { (response_function)(self.respond_handle, size, ptr) } {
            lv2_sys::LV2_Worker_Status_LV2_WORKER_SUCCESS => Ok(()),
            lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_UNKNOWN => Err(WorkerError::Unknown),
            lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_NO_SPACE => Err(WorkerError::NoSpace),
            _ => Err(WorkerError::Unknown),
        }
```

Reply:
Seems i haven't a clear idea of what is unsafe or not. In particular, my `ManuallyDrop` manipulation are "safe rust" even if it's dangerous, it's confuse me a bit.

It's now solved in my local work